CMakeLists.txt

```diff
@@ -201,16 +203,54 @@ set(CC_SOURCES
 
 add_library(sptk STATIC ${CC_SOURCES})
 target_include_directories(sptk PUBLIC
-  ${PROJECT_SOURCE_DIR}/include
-  ${THIRD_PARTY_DIR}
-  ${THIRD_PARTY_DIR}/REAPER
-  ${THIRD_PARTY_DIR}/WORLD
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<BUILD_INTERFACE:${THIRD_PARTY_DIR}>
+  $<BUILD_INTERFACE:${THIRD_PARTY_DIR}/REAPER>
+  $<BUILD_INTERFACE:${THIRD_PARTY_DIR}/WORLD>
   )
+
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+
 install(TARGETS sptk
-  ARCHIVE DESTINATION lib
+  EXPORT ${PROJECT_NAME}Targets
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
   )
+
 install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/SPTK
-  DESTINATION include
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+  )
+
+install(EXPORT ${PROJECT_NAME}Targets
+  FILE ${PROJECT_NAME}Targets.cmake
+  NAMESPACE ${PROJECT_NAME}::
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
+  )
+
+file(WRITE ${PROJECT_BINARY_DIR}/Config.cmake.in
+  "@PACKAGE_INIT@\n\ninclude(\"\${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@Targets.cmake\")\n\ncheck_required_components(@PROJECT_NAME@)\n"
```

Comment:
Is it normal to generate Config.cmake.in? Should it put on the repository?

Reply:
I can put it in the repository as a file as well. It makes no difference. Do want another file in the repo?

Reply:
Thank you for changing.